crates/rpc/rpc/src/result.rs

```diff
@@ -125,6 +126,28 @@ impl ToRpcResultExt for RethResult<Option<Block>> {
     }
 }
 
+impl ToRpcResultExt for RethResult<()> {
+    type Ok = ();
+
+    fn map_ok_or_rpc_err(self) -> RpcResult<<Self as ToRpcResultExt>::Ok> {
+        match self {
+            Ok(()) => Ok(()),
+            Err(err) => Err(internal_rpc_err(err.to_string())),
+        }
+    }
+}
+
+impl ToRpcResultExt for Result<SealedBlock, PayloadError> {
+    type Ok = SealedBlock;
+
+    fn map_ok_or_rpc_err(self) -> RpcResult<<Self as ToRpcResultExt>::Ok> {
+        match self {
+            Ok(block) => Ok(block),
+            Err(err) => Err(internal_rpc_err(err.to_string())),
+        }
+    }
+}
```

Comment:
I believe this could be simplified for `<T>`

`impl<T> ToRpcResultExt for RethResult<T> {`